### DISCUSSION TITLE
Libiio release v0.25-rc.1

### DISCUSSION THREAD

**Discussion by pcercuei** [analogdevicesinc/libiio#1014] [Announcements]
## Changes:

* a231814007bb40ef607efea46b59fc16c9020606 dns_sd_windows: Fix discovery on IPv6 once again
* 99cd3acd80a6406e960ea0f5a557375da895b4d6 scanning: add hwmon only systems to scanning.
* 8492a78330126f178e4cc70eb4d03da39b70be46 sync results of local scans and network scans
* f6a452bf4fa2930f6a90c4cdeb615d72c61fc02d dnssd: Fix off by one error when adding device names
* f5ded6afd9efcae08b1845d9945800a05aeb9db5 CI: Enable serial backend on OSX
* 647ff54f05c6af4aba0e94b83aef969726521384 CI: Don't force CMAKE_SYSTEM_NAME for OSX
* a5d9a2a25325dd69c43fa65201359169d0f3e08e python: Rework bindings install
* 3d87084ddd7f949738dbfc3571cf7cf39711044c device: Fix sample size calculation
* a01da8a5008e813d4036783ee86a2b01dbae483a CI: fix macOS tar balls
* c74e32e16fd828b7e26616d94ba5308f42c666c2 bonjour: Fix discovery on IPv6 network
<details><summary><b>See More</b></summary>

* a59036301de63decf7c1738f5a54f316af3a695d dns_sd_windows: Append interface number to IPv6 link-local addresses
* 007d10a3eb38e7a47ac4e23ff5b53758bc9227a2 iio.h: Document URI for IPv6 addresses with interface name
* 79ae05b3efe7bc8b15351bc4a5e784148dfdf51c network: Fix read-only strings being overwritten
* 8904ef7d8c84c568f552277661ecb2e04fc03e91 avahi: Fix detection of contexts at IPv6 addresses
* eebff52af4154d56e85afabc19b707cb8f2891b3 mdns: update windows to properly timeout
* 209163ae9ca4d23a8078c179b2dd1971d4e6f878 utilities.c: Add utility function iio_read_counter_us()
* 879e49b976748e08639ca9ebe88a628c1633eed0 CI: fix macOS checkout
* c34084e59d2412e42b991cb89a8649258fa016a3 dns_sd_windows: Fix invalid printf format string
* eee8cba6f42177a094ebe16759606a719676fda7 CMake: Bump version to v0.25
* ee1d39c6ea0e33fb6c60d046f2d47a0261f3f7be CI: make file executable
* 405204ec01930bd2a05fd1e47e054ff4be55f7f2 CI: update artifacts
* baf2d391f2214c4b03d6699d2bb63c98dae73cc6 tests/examples: Fix non-compliant function declarations
* 3346657f02993e2827368d815d8c9fde14ccc861 CI: add macOS-13 arm64 build
* 4b71f072abf0e21d868e2277f6be93165cbf3010 Add rpathing and libusb updates for macOS CI
* 9cc2e10f6ec165a594d4780887f99362740c8ad0 CI:add MinGW build
* 25714d1ffcdb6f503d6a1de1d6c4f88549e81c5e network: fix snprintf format string for MinGW
* 28e1b304bc3e10cec7d2aa70e1de2eb60dc68bb6 CMake: Find libusb include dir using pkg-config
* ad199c63b0cf2e0fcfde001f118b0d610dcd1fb3 IIOD: Better handle errors of iio_buffer_{refill,push} [ #981 ]
* 41d14d4373d8452416e560ecd0e55108dcc68395 update packaging: add requirements into debian tar.gz files
* 7ed5ed7e1f9e80e02520aa60ccc1f68644975826 packaging: update DEB_DETECT_DEPENDENCIES
* 5116cc0aa2753e840cc14d72691c9a6a5922e9e1 codespell: ignore 'mis'
* 2a361bc049149852bd19d5502542d2c5190ba70c README_BUILD.md: Fix Github not displaying array properly
* 1032c8275ad415309c281d9640087f5dfd95c850 README_BUILD: Add LOG_LEVEL to the doc
* cb3bcf6220f52bbc99c3d1fe41681946d5f38a4b cmake: Fix invalid Provides: field in Debian packages
* 8ab3029cf7e8afc932a70a5a824fd88638b43175 cmake: Handle conflict between generated .deb and Debian package [ #923 ]
* 508e4bd99d9c3bcfb1d54fae2f8ab31902aa17a6 CI:fix macOS tar balls
* 8739c16973bd3efc1759894e2d75ce46a277ba43 spelling: Ignore some false spelling errors
* 53d94214d629e3f9a0835aab4febb2034cac631e Fix random minor spelling errors
* 7a1269e12a4ffeb75d88f30c856c8158448bf4d9 CI: Add a spelling check into CI
* 739627532b59279d11869745be49894d9cc07d82 local: change exclusionary language
* 0b14634aa7c2efc36d35b041c5511c93eb6a0c0f cmake: turn HWMON on by default, and SERIAL ON for CI
* 1753dbc3840cbc3a2c0b3e300b0a64e130f0c10d iiod: Re-create IIO context when receiving USR1 signal
* f5f8422559a08ba44a409a4f97fc11bdfb3e2b2a iiod: thread_pool: Add function thread_pool_restart()
* e26bff111f4c7e37222ab34014cde3f9c6aab31f iiod: thread_pool: Add helper thread_pool_purge_events()
* 43f03aca5d9bce54c91dadbf8d93538d9344be73 iiopp: added toggle_iio_feature for C++ bindings
* e89543607eeaddd3f8d12a8018d565ffa6e6443f iiopp: Fixed typo ContextInfo::description()
* ac26a23d2636b561f0b22230880be227938d8f6f html: Minor update about C++
* 5506edc662c0ef860ff709e4a43a3262a81bf4fd iiopp: Relax required standard to C++11 (requires Boost for optional-type)
* c3336470788ed31670b578237fd49f2923c15df6 iiopp: exceptions now have a specific type, derived from std::system_error
* 82e9508da7a71833088eb5a8302f944d76020f44 iiopp: Fixed typo in documentation
* 9a869572d863ca8e26ed5d6ea49faae4d7766d76 CI:add new builds
* 690ab5a7ce3841197905f8dd42f78083ca9cb7ff Add C++ API
* cf1c0728d0ecf874b88b2a3edcdb10e11b8c00c0 utils: add 'version' option to each util
* 0a5eaa2b745a280b889a7014afa4418ddcaac6d1 add uninstall target [ #694 ]
* bde61deac34ffbb6e6e50a7b6c02c69b0d5edb0a cmake: set the min version to 2.8.12
* 7cb06e0f4e09208852ad464545003c8eb0b6e2e8 treat warnings as errors in CI
* 33b8b70c990b3f0dcdde5cc7f95ed337175e61f6 man: iio_{read,write}dev: Update examples
* eacddbffd995d608862d84f12199756aa64cabd5 doc: Generate hyperlinks to manuals of utilities
* 459d815f81a569017b5b41c5e6b1a558949e3ec2 man: Add missing man pages
* d8d66ef5127bec624f511fefe1372b49d6f67995 man: Simplify call to make_man.sh
* 9f4949fa1b24a72174a3dd635554e828826f9672 man: manage common parts of man pages in one place
* 800c3946dddd24a3ee1acb5d2cbedacd4fb34650 tests: Move iio_adi_xflow_check to examples
* fe21df7a138f496bf3890c7cd77b4861ed1feb8b tests: Change -T/--Timeout option to -d/--duration in iio_stresstest
* 1a19da1eeee848d24357a3de1af93e3d7626e14e Update Doxyfile.in files
* ca72c7dbc92cbee20fc1af66f9973ec969a9ce30 update Contributors file
* 8e0c83c7ab69a849c5b99cef15843d102093dee1 CI: make sure build instructions are kept up to date
* 642e033e26017e0b0ad5465a328bd24e229df1d4 README_BUILD.md: update with a few settings from cmake files
* 9128e9fffd281b47f5556f4dae29d9e39fbe26ba channel: Don't lookup attributes in special case of reading all attrs
* b36b19db32a2db5f9b4c48eba0c8f9b1e7231446 cmake: tell people if we are/are not install udev rules
* f1fd3cef150471f223f64c7469534c9abcb7d4af CI:fix ARM error
* eef0f7775e0820bb2f7309a3a2c2edc4dd0431a1 set errno to 0, and check after calling library functions
* a037c83aa580b5f9f7d28db6b99602b53d18089d Distribution.xml: Make libiio.pkg installable without using Rosetta
* 8eda862d03fa95125482db9eb30f138f196d71d4 utils: catch calloc memory failures in iio_stresstest
* 91e5f979a672ece12aa230c55b5658c65d1c9b14 channel: Support attribute names that start with the channel's label
* c61f45095a173388178c159897b1f0303b8affb7 tests: don't dereference a var before checking it.
* 6136f11d054bd3f94c99fec6a13b785d919c8cc2 CMake: Sanitize IIO feature list
* c0f55139d0b46998af5d6f36027570cf8ede138a CI: remove builds with deprecated version
* fdecc3622738580f047fbda0c265d86e06175691 usb: don't error during a scan on linux when there are no usb hosts. [ #925 ]
* daddd3f98795e2e0abb53eb586b188a129b3e801 CI: Fix macOS_11
* fb0d6c16cbfce12e9e1df3131b1d8f1935c78ab9 tests: License all files to GPLv2+
* 9938d015c667e33e6afd54e5645f0c4039093736 Clarify licenses, round 3
* 29d592ec40de5b8e4a92546d8bd1eb095200717e serial: Properly flush the input buffer
* 2945479041f6f8df828f81f91de88d4271e0d588 Clarify licenses, round 2
* 4d2e07553c553417a3354ce69b352430d5c16640 Clarify licenses
* a08ad5fa4d67fed641aebef40c955dde4a9b5ebe CMake: Fix detection of libxml2
* 4e577caba1e1b3717ac5efb542327406947fbb6f CI:fix sphinx-build error
* d05e815b0162ba5d6959a89aea91539f6cb8a4d2 CI:set Python to use latest 3.x version
* e598e4b9be84dc299c5ab9fb0da4d3cc29182030 iio.h: sync to upstream kernels
* c889bba9b9fda18178ca75108b5ba7cd845f1ccf CI: add iio_event_type to kernel side checking
* 1246cb031728f4958cfb4324678f600e637bf21c Add missing c_uint type for several python/c bindings
* cee2d1364b12221b37ede3591bb37a623c7b313e CI:fix sphinx error
* 40ffc5d9b8f6e47866292bf1365ac3a5d22cf123 iiod/serial.c: fix sparc build
* 4a869116f0bad8d669319611c0ebb4596a33e2a3 CI:update README.md and other files for artifacts
* cee1e4f0bfcf067ae384149698d1c293d150a6af CI:update ci
* 5b46fa132222efa1feccc32abf8aae8536c1a1c9 CI:add kernel check [ #891 ]
* d23cbaf997d0da18c83c4b4199ab2bc7ba906b63 ci: Build man pages as part of documentation
* 43dd6dea0ea0267d7de388e0f3f682be731df59d Disable libzstd for macos builds

This list of changes was [auto generated](https://dev.azure.com/AnalogDevices/OpenSource/_build/results?buildId=27622&view=logs).</details>

<hr /><em>This discussion was created from the release <a href='https://github.com/analogdevicesinc/libiio/releases/tag/v0.25-rc.1'>Libiio release v0.25-rc.1</a>.</em>

**Comment by Jordanx21** [comment]
Hello,
the folders on windows.zip are empty. Could you please add all the files on it?
Thanks!

**Reply by pcercuei** [reply]
@Jordanx21 Yes, this was reported a few days ago and fixed in the master branch, so  the real v0.25 release will have the files.
In the meantime you can download the files directly from Azure, https://dev.azure.com/AnalogDevices/OpenSource/_build/results?buildId=27898&view=artifacts&pathAsName=false&type=publishedArtifacts
